app/models/course.rb

```diff
@@ -6,4 +6,13 @@ class Course < ActiveRecord::Base
   belongs_to :creator, class_name: User.name
   has_many :course_users, inverse_of: :course, dependent: :destroy
   has_many :users, through: :course_users
+
+  before_validation :set_default_values, on: :create
```

Comment:
OK,I searched for a while, you are right. but I think the <code>on</code> clause is still needed, because there is no reason to call this function after the record is found..